engine/lib/comments.php

```diff
@@ -312,6 +317,145 @@ function _elgg_comments_access_sync($event, $type, $entity) {
 }
 
 /**
+ * Add the owner of the content being commented on to the subscribers
+ *
+ * @param string $hook        'get'
+ * @param string $type        'subscribers'
+ * @param array  $returnvalue current subscribers
+ * @param array  $params      supplied params
+ *
+ * @return void|array
+ *
+ * @access private
+ */
+function _elgg_comments_add_content_owner_to_subscriptions($hook, $type, $returnvalue, $params) {
+	
+	$event = elgg_extract('event', $params);
+	if (!($event instanceof \Elgg\Notifications\Event)) {
+		return;
+	}
+	
+	$object = $event->getObject();
+	if (!elgg_instanceof($object, 'object', 'comment')) {
+		// can't use instanceof ElggComment as discussion replies inherit
+		return;
+	}
+	
+	$content_owner = $object->getContainerEntity()->getOwnerEntity();
+	if (!($content_owner instanceof ElggUser)) {
+		return;
+	}
+	
+	$notification_settings = get_user_notification_settings($content_owner->getGUID());
+	if (empty($notification_settings)) {
+		return;
+	}
+	
+	$returnvalue[$content_owner->getGUID()] = [];
+	foreach ($notification_settings as $method => $enabled) {
+		if (empty($enabled)) {
+			continue;
+		}
+		
+		$returnvalue[$content_owner->getGUID()][] = $method;
+	}
+	
+	return $returnvalue;
+}
+
+/**
+ * Set the notification message for the owner of the content being commented on
+ *
+ * @param string                           $hook        'prepare'
+ * @param string                           $type        'notification:create:object:comment'
+ * @param \Elgg\Notifications\Notification $returnvalue current notification message
+ * @param array                            $params      supplied params
+ *
+ * @return void|\Elgg\Notifications\Notification
+ *
+ * @access private
+ */
+function _elgg_comments_prepare_content_owner_notification($hook, $type, $returnvalue, $params) {
+	
+	$comment = elgg_extract('object', $params);
+	if (!elgg_instanceof($comment, 'object', 'comment')) {
+		// can't use instanceof ElggComment as discussion replies inherit
+		return;
+	}
+	
+	/* @var $content \ElggEntity */
+	$content = $comment->getContainerEntity();
+	$recipient = elgg_extract('recipient', $params);
+	if ($content->getOwnerGUID() !== $recipient->getGUID()) {
```

Comment:
If the container can't be loaded, this will be fatal. It's a practical reason I like `->guid` over `->getGUID()`.